Network/Socket/Types.hsc

```diff
@@ -79,23 +72,15 @@ import Foreign.Storable
 import Foreign.Marshal.Array
 #endif
 
--- | Type for a socket. Use 'mkSocket' to create 'Socket'.
---   This is typically created by 'socket', 'accept' and 'socketPair'.
-data Socket = Socket
-  {
-    socketFd       :: CInt              -- ^ File descriptor.
-  , socketFamily   :: Family            -- ^ Address family.
-  , socketType     :: SocketType        -- ^ Socket type.
-  , socketProtocol :: ProtocolNumber    -- ^ Protocol number.
-  , socketStatus   :: MVar SocketStatus -- ^ Socket status.
-  } deriving Typeable
+-----------------------------------------------------------------------------
 
-instance Eq Socket where
-  s1 == s2 = socketStatus s1 == socketStatus s2
+-- | Basic type for a socket.
+type Socket = CInt
```

Comment:
My initial reaction to this type is that we probably want a `newtype` not just an alias.

Reply:
I initially wrote `newtype Socket = Socket CInt` and created the `NetworkSocket` typeclass.
All APIs relating to `Socket` had `NetworkSocket s =>`. But I felt it is lengthy. So, I use the synonym.

I experienced that some packages make `Socket` an instance of a class. Haskell 2010 does not allow a synonym to be an instance. However, `FlexibleInstances` solves this. So, `type Socket = CInt` is good enough for me so far.

But I'm neutral on this issue.

Reply:
Oh. We don't have to provide `NetworkSocket` because nobody wants to extend `Socket`.
Providing `mkSocket :: CInt -> Socket` is good enough. OK. I will change this.

Reply:
OK. This is done.